// ng-intercom/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var intercom_1 = require("./intercom/intercom");
exports.Intercom = intercom_1.Intercom;
var hide_directive_1 = require("./directives/hide.directive");
exports.IntercomHideDirective = hide_directive_1.IntercomHideDirective;
var intercom_module_1 = require("./intercom.module");
exports.IntercomModule = intercom_module_1.IntercomModule;
var show_directive_1 = require("./directives/show.directive");
exports.IntercomShowDirective = show_directive_1.IntercomShowDirective;
var show_messages_directive_1 = require("./directives/show-messages.directive");
exports.IntercomShowMessagesDirective = show_messages_directive_1.IntercomShowMessagesDirective;
var show_new_message_directive_1 = require("./directives/show-new-message.directive");
exports.IntercomShowNewMessageDirective = show_new_message_directive_1.IntercomShowNewMessageDirective;
var shutdown_directive_1 = require("./directives/shutdown.directive");
exports.IntercomShutdownDirective = shutdown_directive_1.IntercomShutdownDirective;
var track_event_directive_1 = require("./directives/track-event.directive");
exports.IntercomTrackEventDirective = track_event_directive_1.IntercomTrackEventDirective;
var leads_1 = require("./leads/leads");
exports.Leads = leads_1.Leads;
