// ng-intercom/dist/intercom.module.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var config_token_1 = require("./shared/config-token");
var http_1 = require("@angular/common/http");
var intercom_1 = require("./intercom/intercom");
var hide_directive_1 = require("./directives/hide.directive");
var show_directive_1 = require("./directives/show.directive");
var show_messages_directive_1 = require("./directives/show-messages.directive");
var show_new_message_directive_1 = require("./directives/show-new-message.directive");
var shutdown_directive_1 = require("./directives/shutdown.directive");
var track_event_directive_1 = require("./directives/track-event.directive");
var leads_1 = require("./leads/leads");
var options_1 = require("./shared/options");
var router_1 = require("@angular/router");
var IntercomModule = (function () {
    function IntercomModule() {
    }
    IntercomModule.forRoot = function (config) {
        return {
            ngModule: IntercomModule,
            providers: [
                { provide: config_token_1.CONFIG, useValue: config },
                intercom_1.Intercom,
                leads_1.Leads,
                options_1.Options
            ]
        };
    };
    return IntercomModule;
}());
IntercomModule.decorators = [
    { type: core_1.NgModule, args: [{
                imports: [
                    http_1.HttpClientModule,
                    router_1.RouterModule
                ],
                declarations: [
                    hide_directive_1.IntercomHideDirective,
                    show_messages_directive_1.IntercomShowMessagesDirective,
                    show_new_message_directive_1.IntercomShowNewMessageDirective,
                    show_directive_1.IntercomShowDirective,
                    shutdown_directive_1.IntercomShutdownDirective,
                    track_event_directive_1.IntercomTrackEventDirective
                ],
                exports: [
                    hide_directive_1.IntercomHideDirective,
                    show_messages_directive_1.IntercomShowMessagesDirective,
                    show_new_message_directive_1.IntercomShowNewMessageDirective,
                    show_directive_1.IntercomShowDirective,
                    shutdown_directive_1.IntercomShutdownDirective,
                    track_event_directive_1.IntercomTrackEventDirective
                ]
            },] },
];
/** @nocollapse */
IntercomModule.ctorParameters = function () { return []; };
exports.IntercomModule = IntercomModule;
